src/components/Lesson.vue

```diff
@@ -258,11 +262,27 @@ export default {
         return new IPFS({repo: Math.random().toString()})
       }
     },
+    clearCache: function () {
+      let cacheKey = this.$attrs.exercise + this.$attrs.lessonTitle
+      delete localStorage[cacheKey]
+      this.cachedCode = false
+      if (this.code !== this.$attrs.code) {
+        this.code = this.$attrs.code || defaultCode
+        this.editor.setValue(this.code)
+      }
+    },
     onMounted: function (editor) {
       this.editor = editor
     },
     onCodeChange: function (editor) {
-      // console.log(editor.getValue())
+      if (editor.getValue() === this.$attrs.code) {
+        this.clearCache()
+      } else {
+        let cacheKey = this.$attrs.exercise + this.$attrs.lessonTitle
+        localStorage[cacheKey] = editor.getValue()
+        this.code = editor.getValue()
+        this.cachedCode = true
```

Comment:
Perhaps we can set cachedCode to some kind of string so that we can have different text when the code is loaded from cache and when we've *just* started saving it?

Reply:
Done!